### PR TITLE
docs: fix link to metadata registry file; update TOC

### PR DESCRIPTION
**Pull request by mohanraj-r**
### What does this PR do?
- Fixes broken link to metadata registry file in Handbook
- Updates TOC in Handbook
- AUTO adds `DataImportManagementSettings` to Metadata support doc

### What issues does this PR fix or reference?

@W-10850157@

### CODE REVIEW

File: METADATA_SUPPORT.md [hunk 1]
```diff
@@ -489,6 +489,7 @@ v55 introduces the following new types.  Here's their current level of support
 |ConvReasonReportDefinition|❌|Not supported, but support could be added|
 |ConvReasonReportSegmentDef|❌|Not supported, but support could be added|
 |CustomAddressFieldSettings|✅||
+|DataImportManagementSettings|✅||
```
Comment (mohanraj-r) [inline METADATA_SUPPORT.md:492]: @mshanemc Guess this got auto-updated? Should I keep/revert this?

**Review by jeffb-sfdc — APPROVED** [review]

**Review by gbockus-sf — APPROVED** [review]
👍

File: METADATA_SUPPORT.md (hunk 1 above)
Comment (mohanraj-r) [reply METADATA_SUPPORT.md:492]: I will revert this change just to be safe

**Review by WillieRuemmele — APPROVED** [review]

File: METADATA_SUPPORT.md (hunk 1 above)
Comment (gbockus-sf) [reply METADATA_SUPPORT.md:492]: yeah I saw this on one of my PRs as well (not this specific change, but the same pattern).  We should make a note to understand why it does this at some point :)

Comment (WillieRuemmele) [reply METADATA_SUPPORT.md:492]: there's an automated script that will pull the latest metadata types and update the registry, I think it's a `pre-push` hook

Comment (mohanraj-r) [reply METADATA_SUPPORT.md:492]: Thanks, not reverting this add then as it seems to be have been added by design

File: contributing/metadata.md [hunk 2]
```diff
@@ -2,6 +2,14 @@
 
 You can view the existing metadata coverage by release using [METADATA_SUPPORT.md](../METADATA_SUPPORT.md)
 
+- It can be updated by running
```
Comment (mohanraj-r) [inline contributing/metadata.md:5]: @WillieRuemmele @mshanemc Can you review the following new changes please.

**Review by mshanemc — APPROVED** [review]

**Pull request merged**